lambda/audit.py

```diff
@@ -0,0 +1,319 @@
+        event = {
+            "type": "OrganizationTeamCount",
+            "org": org,
+            "count": len(teams),
+            "audit_id": audit_id,
+        }
+        LOG.info(event)
+        for team in teams:
+            event = {
+                "type": "OrganizationTeam",
+                "org": org,
+                "member": None,
+                "team": team,
+                "repository": None,
+                "audit_id": audit_id,
+            }
+            LOG.info(event)
+            sns_team_membership = create_sns_message(
+                {
+                    "action": "log_org_team_membership",
+                    "org": org,
+                    "team": team,
+                    "audit_id": audit_id,
+                }
+            )
+            publish_alert(sns_team_membership)
+
+            sns_team_repos = create_sns_message(
+                {
+                    "action": "log_org_team_repos",
+                    "org": org,
+                    "team": team,
+                    "audit_id": audit_id,
+                }
+            )
+            publish_alert(sns_team_repos)
+    return True
+
+
+def log_org_team_membership(message=None):
+    """ Audit github organization team membership """
+    org = os.environ.get("GITHUB_ORG")
+    team = message.get("team", None)
+    team_name = team["slug"]
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info({"action": "Audit team members", "org": org, "team": team_name})
+        if team is not None:
+            members = github_api.get_github_org_team_members(org, team_name)
+            event = {
+                "type": "OrganizationTeamMemberCount",
+                "org": org,
+                "team": team,
+                "count": len(members),
+                "audit_id": audit_id,
+            }
+            LOG.info(event)
+            for member in members:
+                event = {
+                    "type": "OrganizationTeamMember",
+                    "org": org,
+                    "member": member,
+                    "team": team,
+                    "repository": None,
+                    "audit_id": audit_id,
+                }
+                LOG.info(event)
+        else:
+            LOG.error({"error": "Team not specified"})
+
+    return True
+
+
+def log_org_team_repos(message=None):
+    """ Audit github organization repo team access """
+    org = os.environ.get("GITHUB_ORG")
+    team = message.get("team", None)
+    team_name = team["slug"]
+    audit_id = message.get("audit_id")
+    if audit_id is not None:
+        LOG.info(
+            {
+                "action": "Audit organization team repos",
+                "org": org,
+                "team": team_name,
+                "audit_id": audit_id,
+            }
+        )
+        if team is not None:
+            repos = github_api.get_github_org_team_repositories(org, team_name)
```

Comment:
Can a repository be owned by more than one team?

Reply:
Yes this is about who has access - not who has ownership.
As an example when someone leaves you can list all the repositories they have access to whether via being a member of a team or by being a named contributor. So we can ensure all their access is revoked. 

Obvs. that relies on being able to match their username back to a work email but that's another issue.

Reply:
Thanks for clarifying.